api/src/modules/scenario-interventions/services/intervention-generator.service.ts

```diff
@@ -0,0 +1,157 @@
+import { Injectable } from '@nestjs/common';
+import { CreateScenarioInterventionDto } from 'modules/scenario-interventions/dto/create.scenario-intervention.dto';
+import {
+  SCENARIO_INTERVENTION_TYPE,
+  ScenarioIntervention,
+} from 'modules/scenario-interventions/scenario-intervention.entity';
+import { MaterialsService } from 'modules/materials/materials.service';
+import { BusinessUnitsService } from 'modules/business-units/business-units.service';
+import { AdminRegionsService } from 'modules/admin-regions/admin-regions.service';
+import { SuppliersService } from 'modules/suppliers/suppliers.service';
+import { SourcingLocation } from 'modules/sourcing-locations/sourcing-location.entity';
+import { SourcingData } from 'modules/import-data/sourcing-data/dto-processor.service';
+
+/**
+ * @description: Utility wrapper to encapsulate some logics regarding new intervention generation
+ */
+
+@Injectable()
+export class InterventionGeneratorService {
```

Comment:
While I agree that that the original createScenarioIntervention function in ScenarioIntervention service was too big, and needed some functionality extracted, I think the approach could be slightly different than using another service/helper class, but not without deeper analysis for a bigger refactoring.
I would suggest using a Strategy pattern or similar as a basis for it, as it seems there's quite a bit of base code execution between each type.
In any case the current solution, makes the code more readable, and abstracts away some code, so it's fine for now :)

Reply:
I do totally agree that the whole workflow needs a deeper analysis for a proper refactoring. I'm all ears about what you might suggest for this. 
As you mentioned , you could push forward with this for the time being, but adding reviewing this as huge bullet point on our tech debt list

Let me know when you'd like to invest some time on this, as I think it will pay off